cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORGenerator.java

```diff
@@ -402,9 +417,20 @@ public final void writeStringField(String fieldName, String value)
             return;
         }
         _verifyValueWrite("write String value");
+        _decrementList();
         _writeString(value);
     }
 
+	
+    public final void writeFieldLong(long size) throws IOException {
```

Comment:
I don't think annotations would be needed, as type information itself is available wrt Maps (as POJO fields always have String names). So it'd be adding new methods in `JsonGenerator` (and possibly `JsonParser`), then wiring.
But realistically speaking I don't think this would be implemented by core team any time soon.